### PR TITLE
feat(atomic,headless): add recommendations component

### PR DESCRIPTION
**Pull request by fbeaudoincoveo**
https://coveord.atlassian.net/browse/CDX-1562

Also allow the initial state of the cart to be set when initializing the headless commerce engine.

### CODE REVIEW

File: packages/atomic/src/pages/examples/commerce-website/engine.mjs [hunk 1]
```diff
@@ -25,55 +20,36 @@ export const commerceEngineConfig = {
...
-        return `<span><a href='${product.clickUri}'><img width="100px" src='${product.ec_images[0]}'>${product.ec_name}</a></span>`;
-      })
-      .join('');
-
-  return products
-    .map((product) => {
-      return `<div>${product.name}</div>`;
-    })
-    .join('');
+  cart: {
```
Comment (Spuffynism) [inline packages/atomic/src/pages/examples/commerce-website/engine.mjs:23]: After chatting with @samisayegh, @sallainCoveo and @BenKandelaars, we'll want to change the cart's state structure to better support Coveo's needs.

Not blocking but just a heads up 😄 You should get a meeting invite about this soon 😅

File: packages/atomic/src/pages/examples/commerce-website/homepage.html [hunk 2]
```diff
@@ -9,43 +9,70 @@
...
+                  <atomic-product-text field="ec_brand"></atomic-product-text>
+                </atomic-field-condition>
+
+                <atomic-field-condition class="field" if-defined="ec_rating">
+                  <atomic-result-rating field="ec_rating"></atomic-result-rating>
+                </atomic-field-condition>
+
+                <atomic-field-condition class="mt-4 text-2xl font-bold field" if-defined="ec_price">
+                  <atomic-result-number field="ec_price">
+                    <atomic-format-currency currency="CAD"></atomic-format-currency>
```
Comment (Spuffynism) [inline packages/atomic/src/pages/examples/commerce-website/homepage.html:57]: Can this component use whatever is set on the commerce engine initialization as currency? https://github.com/coveo/ui-kit/blob/master/packages/headless/src/integration-tests/commerce.test.ts#L38

File: packages/headless/src/api/commerce/common/product.ts [hunk 3]
```diff
@@ -83,6 +82,18 @@ export interface Product {
...
    */
   ec_rating?: number;
+  /**
+   * The gender the product is intended for.
+   */
+  ec_gender?: string;
+  /**
+   * The listing page the product was returned in.
+   */
+  ec_listing?: string;
```
Comment (Spuffynism) [inline packages/headless/src/api/commerce/common/product.ts:92]: I don't think this is actually used anymore. 🤔 @anthonydelage ?

**Review by Spuffynism — APPROVED** [review]
@BenKandelaars adding you as reviewer because of your work in recs

File: packages/atomic/src/pages/examples/commerce-website/homepage.html (hunk 2 above)
Comment (fbeaudoincoveo) [reply packages/atomic/src/pages/examples/commerce-website/homepage.html:57]: Good point!

Created [KIT-3172](https://coveord.atlassian.net/browse/KIT-3172)

File: packages/atomic/src/pages/examples/commerce-website/homepage.html [hunk 4]
```diff
@@ -9,43 +9,70 @@
...
+                  </atomic-result-number>
+                </atomic-field-condition>
+                <atomic-product-description></atomic-product-description>
+                <atomic-result-fields-list>
```
Comment (olamothe) [inline packages/atomic/src/pages/examples/commerce-website/homepage.html:61]: Same as above: In general I think it's much better to *not* add any result template component here, and wait for their commerce counterpart

**Review by olamothe — APPROVED** [review]

File: packages/headless/src/api/commerce/common/product.ts (hunk 3 above)
Comment (fbeaudoincoveo) [reply packages/headless/src/api/commerce/common/product.ts:92]: I'll remove it... easier to add stuff than removing it once the sub-package is GA

**Pull request merged**